### ISSUE TITLE
Gradients do not exist for variables when applying learnable adjacency matrix to the GATConv layer

### ISSUE THREAD

**Issue by zzzy0828** [danielegrattarola/spektral#395]
Hi!
Recently I bulid learnable adjacency matrices for modeling in which case connections between different sites vary at different time steps. (Each time step corresponds to a specific graph structure). I used several fully connected layers (Dense) to make the initial adjacency matrices learnable, where parameters can be updated through model training, namely gradient descent. However, when applying the learnable adjacency matrices to the GATConv layer, Tensorflow Warnings appear as follows:
****
WARNING:tensorflow:Gradients do not exist for variables ['dense_2/kernel:0', 'dense_4/kernel:0', 'dense_3/kernel:0', 'dense_5/kernel:0'] when minimizing the loss.
WARNING:tensorflow:Gradients do not exist for variables ['dense_2/kernel:0', 'dense_4/kernel:0', 'dense_3/kernel:0', 'dense_5/kernel:0'] when minimizing the loss.
WARNING:tensorflow:Gradients do not exist for variables ['dense_2/kernel:0', 'dense_4/kernel:0', 'dense_3/kernel:0', 'dense_5/kernel:0'] when minimizing the loss.
WARNING:tensorflow:Gradients do not exist for variables ['dense_2/kernel:0', 'dense_4/kernel:0', 'dense_3/kernel:0', 'dense_5/kernel:0'] when minimizing the loss.
****
If I use the GCNConv layer rather than the GATConv layer, Warnings above disappeared. 
I guess that the adjacency matrices didn't truly take part in the process of graph attention. Is that true?
How can I handle this problem (avoid these warnings) when using a GATConv layer?
Expect for your reply, thank you very much!

**Comment by danielegrattarola** [comment]
GAT computes the adjacency matrix as a function of the node features so there might be something unexpected going on there. 
The warnings seem to indicate that your learnable weights are not being used to compute the actual adjacency matrix.

Can you post a snippet to reproduce the issue?
Thanks

**Comment by danielegrattarola** [comment]
Yeah unfortunately GAT is not designed to work like that, there is no way to backpropagate the error to the dense layer that's generating the adjacency matrix.
GCNConv is a better choice.

**Comment by zzzy0828** [comment]
OK, I got it.
Thank you for your reply.
Thank you very much!
Best wishes.
